inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -941,6 +941,39 @@ def normalize_collaborations(obj, eng):
     return obj
 
 
+def _get_correct_affiliation_data(affiliations):
+    cleaned_affiliations = []
+    for aff in affiliations:
+        cleaned_affiliations.append({key: val for key, val in aff.items() if key in ['value', 'record']})
+    return cleaned_affiliations
+
+
+def _extract_matched_aff_from_highlight(highlighted_raw_affs, author_raw_affs, author_affs):
+    highlight_matches = [re.finditer('(?=<em>).*?(?<=<\/em>)', raw_aff) for raw_aff in highlighted_raw_affs]
+    raw_aff_highlight_lenghts = [sum([match.span()[1] - match.span()[0] for match in aff_matches]) for aff_matches in highlight_matches]
+    longest_highlight_idx = raw_aff_highlight_lenghts.index(max(raw_aff_highlight_lenghts))
+    extracted_raw_aff = re.sub('<em>|</em>', "", highlighted_raw_affs[longest_highlight_idx])
+    for ix, raw_aff in enumerate(author_raw_affs):
+        if raw_aff['value'] == extracted_raw_aff:
+            return [author_affs[ix]]
+
+
+def _find_unambiguous_affiliation(result):
+    for matched_author in result:
+        matched_author_data = matched_author.meta.inner_hits.authors.hits[0].to_dict()
+        matched_author_raw_affs = matched_author_data['raw_affiliations']
+        matched_author_affs = matched_author_data['affiliations']
+        if len(matched_author_raw_affs) == 1:
+            return _get_correct_affiliation_data(matched_author_affs)
+        elif len(matched_author_raw_affs) == len(matched_author_affs):
+            matched_aff = _extract_matched_aff_from_highlight(
+                matched_author.meta.highlight['authors.raw_affiliations.value'],
+                matched_author_raw_affs,
+                matched_author_affs
+            )
+            return _get_correct_affiliation_data(matched_aff)
```

Comment:
you're calling this function in every branch, better take it out of this function. Also this should be called `_clean_up_affiliation_data` to make its purpose more clear.